kk/views/base.py

```diff
@@ -33,3 +36,24 @@ def get_url(self, obj):
             url = request.build_absolute_uri(url)
 
         return url
+
+
+class BaseVoteViewSet(object):
+    """
+    Base view for types, for which vote can be added.
+    """
+
+    @detail_route(methods=['post'])
+    def votes(self, request, **kwargs):
+        comment = self.get_object()
+
+        # Check if user voted already. If yes, return 400.
```

Comment:
I don't think this really warrants a 400. Maybe a 304 (Not Modified), since no change is done if the user has voted already.